bedrock/firefox/redirects.py

```diff
@@ -2,66 +2,12 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+import re
+
 from bedrock.redirects.util import mobile_app_redirector, no_redirect, platform_redirector, redirect
 
-PRODUCT_OPTIONS = ["firefox", "focus", "klar"]
-CAMPAIGN_OPTIONS = [
-    "firefox-whatsnew",
-    "firefox-welcome-4",
-    "firefox-welcome-6",
-    "firefox-welcome-17-en",
-    "firefox-welcome-17-de",
-    "firefox-welcome-17-fr",
-    "firefox-browsers-mobile-get-app",
-    "firefox-browsers-mobile-focus",
-    "mzaonboardingemail-de",
-    "mzaonboardingemail-fr",
-    "mzaonboardingemail-es",
-    "mzaonboardingemail-it",
-    "mzaonboardingemail-nl",
-    "mzaonboardingemail-pl",
-    "firefox-all",
-    "fxshare1",
-    "fxshare2",
-    "fxshare3",
-    "fxshare4",
-    "fxshare5",
-    "fxshare6",
-    "fxshare7",
-    "fxshare8",
-    "fxshare9",
-    "fxshare10",
-    "fxshare11",
-    "fxshare12",
-    "fxshare13",
-    "fxshare14",
-    "fxshare15",
-    "DESKTOP_FEATURE_CALLOUT_SIGNED_INTO_ACCOUNT.treatment_a",
-    "DESKTOP_FEATURE_CALLOUT_SIGNED_INTO_ACCOUNT.treatment_b",
-    "wnp134-de-a",
-    "wnp134-de-b",
-    "wnp134-de-c",
-    "wnp134-en-ca-a",
-    "wnp134-en-ca-b",
-    "wnp134-en-ca-c",
-    "wnp134-en-na-a",
-    "wnp134-en-na-b",
-    "wnp134-en-na-c",
-    "wnp134-en-uk-a",
-    "wnp134-en-uk-b",
-    "wnp134-en-uk-c",
-    "wnp134-fr-a",
-    "wnp134-fr-b",
-    "wnp134-fr-c",
-    "smi-marvintsp",
-    "smi-koschtaaa",
-    "smi-itsmanjuu",
-    "smi-domiip",
-    "smi-drfaye",
-    "smi-shessdly",
-    "smi-bytereview",
-    "pocket-test",
-]
+# matches only ASCII letters (ignoring case), numbers, dashes, periods, and underscores.
+PARAM_VALUES_RE = re.compile(r"[\w\-\.]+", flags=re.I | re.A)
```

Comment:
I think you can simplify this.
- if you're only dealing with `\w` which is already `[A-Za-z0-9_]` you're already ignoring case, so `re.I` is redundant?
- hyphen and period don't need escaping inside the `[]`
- using the full `re.ASCII` might be nice for those who need a little memory jog on the various flags. :) 

In Python REPL this worked the same for me:
`PARAM_VALUES_RE = re.compile(r"[\w.-]+", flags=re.ASCII)` and is a bit simpler to parse with the brain.

Reply:
For some reason I got an error when running tests when I didn't escape the dash. I'll try it again.

Reply:
oh, it didn't like the dash between the other two, thought it was a "character range". I moved it and it's good now.